src/azul/service/elasticsearch_service.py

```diff
@@ -586,6 +586,16 @@ def transform_request(self,
                 es_response = es_search.execute(ignore_cache=True)
             except elasticsearch.NotFoundError as e:
                 raise IndexNotFoundError(e.info["error"]["index"])
+            except elasticsearch.RequestError as e:
+                if one(e.info['error']['root_cause'])['reason'].startswith('No mapping found for'):
+                    # Index exists but it is empty
+                    if es_search.count() == 0:
```

Comment:
If you look at the implemetaion of .count, it reuses the previous response if it exists. Worse though, even if it round-trips again, it reuses the same query from the original request so this may be conflating "no documents match the filter" with "the index is completely empty".